recipes/qt/6.x.x/conanfile.py

```diff
@@ -545,3 +546,42 @@ def package_info(self):
                 self.cpp_info.frameworks.append("IOKit")     # qtcore requires "_IORegistryEntryCreateCFProperty", "_IOServiceGetMatchingService" and much more which are in "IOKit" framework
                 self.cpp_info.frameworks.append("Cocoa")     # qtcore requires "_OBJC_CLASS_$_NSApplication" and more, which are in "Cocoa" framework
                 self.cpp_info.frameworks.append("Security")  # qtcore requires "_SecRequirementCreateWithString" and more, which are in "Security" framework
+        
+        extension = ""
+        if self.settings.os == "Windows":
+            extension = ".exe"
+        filecontents = "set(QT_CMAKE_EXPORT_NAMESPACE Qt6)\n"
+        filecontents += "set(QT_VERSION_MAJOR %d)\n" % int(self.version.split(".")[0])
+        filecontents += "set(QT_VERSION_MINOR %d)\n" % int(self.version.split(".")[1])
+        filecontents += "set(QT_VERSION_PATCH %d)\n" % int(self.version.split(".")[2])
+        targets = ["moc", "rcc", "tracegen", "cmake_automoc_parser", "qlalr", "qmake"]
+        targets.extend(["qdbuscpp2xml", "qdbusxml2cpp"])
+        if self.options.gui:
+            targets.append("qvkgen")
+        if self.options.widgets:
+            targets.append("uic")
+        if self.options.qttools:
+            targets.extend(["qhelpgenerator", "qtattributionsscanner", "windeployqt"])
+            targest.extend(["lconvert", "lprodump", "lrelease", "lrelease-pro", "lupdate", "lupdate-pro"])
+        if self.options.qtshadertools:
+            targets.append("qsb")
+        if self.options.qtdeclarative:
+            targets.extend(["qmltyperegistrar", "qmlcachegen", "qmllint", "qmlimportscanner"])
+            targets.extend(["qmlformat", "qml", "qmlprofiler", "qmlpreview", "qmltestrunner"])
+        for target in targets:
+            filecontents.append("""if(NOT TARGET ${{QT_CMAKE_EXPORT_NAMESPACE}}::{1})
+    add_executable(${{QT_CMAKE_EXPORT_NAMESPACE}}::{1} IMPORTED)
+    set_target_properties(${{QT_CMAKE_EXPORT_NAMESPACE}}::{1} PROPERTIES IMPORTED_LOCATION ${{CMAKE_CURRENT_LIST_DIR}}/../../../bin/{1}{0})
+endif()
+""".format(extension, target)
```

Comment:
```suggestion
""".format(extension, target))
```
there may be other typos, sorry :(

Reply:
No worries will fix. Like this setup much more complete.